tests/components/decora_wifi/test_config_flow.py

```diff
@@ -0,0 +1,216 @@
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert not result["errors"]
+    # Test that the flow with user input calls setup entry with the appropriate data.
+    with patch(
+        "homeassistant.components.decora_wifi.common.DecoraWiFiSession",
+        side_effect=FakeDecoraWiFiSession,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.Residence",
+        side_effect=FakeDecoraWiFiResidence,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.ResidentialAccount",
+        side_effect=FakeDecoraWiFiResidentialAccount,
+    ), patch(
+        "homeassistant.components.decora_wifi.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.decora_wifi.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        hass.data[DOMAIN] = {}
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+        )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == f"{CONF_TITLE} - {USERNAME}"
+    assert result2["data"] == {
+        CONF_USERNAME: USERNAME,
+        CONF_PASSWORD: PASSWORD,
+    }
+    mock_setup.assert_called_once()
+    mock_setup_entry.assert_called_once()
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+
+
+async def test_reauth_flow(hass: HomeAssistant):
+    """Test the reauth flow."""
+    mock_config = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=USERNAME,
+        data={
+            CONF_USERNAME: USERNAME,
+            CONF_PASSWORD: INCORRECT_PASSWORD,
+        },
+    )
+    mock_config.add_to_hass(hass)
+    # Test that the flow with no user input produces a form.
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "unique_id": mock_config.unique_id,
+        },
+        data=mock_config.data,
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert not result["errors"]
+    # Test reauth with valid login info
+    FakeDecoraWiFiSession.clear_accounts()
+    FakeDecoraWiFiSession.add_account(
+        FakeDecoraWiFiAccount(USERNAME, UPDATED_PASSWORD, switch_models=["D26HD"])
+    )
+    with patch(
+        "homeassistant.components.decora_wifi.common.DecoraWiFiSession",
+        side_effect=FakeDecoraWiFiSession,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.Residence",
+        side_effect=FakeDecoraWiFiResidence,
+    ), patch(
+        "homeassistant.components.decora_wifi.common.ResidentialAccount",
+        side_effect=FakeDecoraWiFiResidentialAccount,
+    ), patch(
+        "homeassistant.components.decora_wifi.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry, patch.dict(
+        hass.data
+    ):
+        hass.data[DOMAIN] = {}
```

Comment:
Please don't access `hass.data[DOMAIN]` in tests

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations